opt: BenchmarkEndToEnd fails in `bench` CI run

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_Tests_Bench/17829837?buildTab=overview&expandBuildProblemsSection=true&hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true
```
[00:37:21] :			 [Bench //pkg/sql/opt/bench:bench_test] --- FAIL: BenchmarkEndToEnd
[00:37:21] :			 [Bench //pkg/sql/opt/bench:bench_test]     test_log_scope.go:165: test logs captured to: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/execroot/com_github_cockroachdb_cockroach/_tmp/4d025bdb26128dab9f5a867b6346d1b0/logBenchmarkEndToEnd76308205
[00:37:21] :			 [Bench //pkg/sql/opt/bench:bench_test]     test_log_scope.go:76: use -show-logs to present logs inline
[00:37:21] :			 [Bench //pkg/sql/opt/bench:bench_test]     bench_test.go:1167: -- test log scope end --
```

Jira issue: CRDB-44684

--- a/pkg/sql/opt/bench/bench_test.go
+++ b/pkg/sql/opt/bench/bench_test.go
@@ -242,7 +242,7 @@ var schemas = []string{
 	`
 	CREATE TABLE json_comp
 	(
-		k INT PRIMARY KEY,
+		k UUID PRIMARY KEY DEFAULT gen_random_uuid(),
 		i INT,
 		j1 JSON,
 		j2 JSON,
@@ -523,10 +523,9 @@ var queries = [...]benchQuery{
 		cleanup: "TRUNCATE TABLE json_table",
 	},
 	{
-		name:    "json-comp-insert",
-		query:   `INSERT INTO json_comp(k, i, j1, j2, j3) VALUES ($1, $2, $3, $4, $5)`,
-		args:    []interface{}{1, 10, `'{"foo": {"bar": {"int": 12345, "int2": 1}}, "baz": false}'`, `'{"str": "hello world"}'`, `'{"c": [2, 3, "baz", true, false, null]}'`},
-		cleanup: "TRUNCATE TABLE json_table",
+		name:  "json-comp-insert",
+		query: `INSERT INTO json_comp(i, j1, j2, j3) VALUES ($1, $2, $3, $4)`,
+		args:  []interface{}{10, `'{"foo": {"bar": {"int": 12345, "int2": 1}}, "baz": false}'`, `'{"str": "hello world"}'`, `'{"c": [2, 3, "baz", true, false, null]}'`},
 	},
 	{
 		name: "batch-insert-one",
@@ -1126,13 +1125,14 @@ func BenchmarkEndToEnd(b *testing.B) {
 	}
 
 	for _, query := range queriesToTest(b) {
+		args := trimSingleQuotes(query.args)
 		b.Run(query.name, func(b *testing.B) {
 			for _, vectorize := range []string{"on", "off"} {
 				b.Run("vectorize="+vectorize, func(b *testing.B) {
 					sr.Exec(b, "SET vectorize="+vectorize)
 					b.Run("Simple", func(b *testing.B) {
 						for i := 0; i < b.N; i++ {
-							sr.Exec(b, query.query, query.args...)
+							sr.Exec(b, query.query, args...)
 							if query.cleanup != "" {
 								sr.Exec(b, query.cleanup)
 							}
@@ -1144,7 +1144,7 @@ func BenchmarkEndToEnd(b *testing.B) {
 							b.Fatalf("%v", err)
 						}
 						for i := 0; i < b.N; i++ {
-							res, err := prepared.Exec(query.args...)
+							res, err := prepared.Exec(args...)
 							if err != nil {
 								b.Fatalf("%v", err)
 							}
@@ -1164,6 +1164,18 @@ func BenchmarkEndToEnd(b *testing.B) {
 			}
 		})
 	}
+}
+
+func trimSingleQuotes(args []interface{}) []interface{} {
+	res := make([]interface{}, len(args))
+	for i, arg := range args {
+		if s, ok := arg.(string); ok {
+			res[i] = strings.Trim(s, "'")
+		} else {
+			res[i] = arg
+		}
+	}
+	return res
 }
 
 var slowQueries = [...]benchQuery{

